components/tidb_query/src/rpn_expr/impl_arithmetic.rs

```diff
@@ -466,30 +461,21 @@ fn int_divide_decimal(
     lhs: &Option<Decimal>,
     rhs: &Option<Decimal>,
 ) -> Result<Option<Int>> {
-    use crate::codec::mysql::Res;
+    let result = try_opt!(arithmetic_with_ctx::<DecimalDivide>(ctx, lhs, rhs)).as_i64();
 
-    if lhs.is_none() || rhs.is_none() {
-        return Ok(None);
-    }
-    let lhs = lhs.as_ref().unwrap();
-    let rhs = rhs.as_ref().unwrap();
-
-    match lhs / rhs {
-        Some(v) => match v {
-            Res::Ok(v) => match v.as_i64() {
-                Res::Ok(v_i64) => Ok(Some(v_i64)),
-                Res::Truncated(v_i64) => Ok(Some(v_i64)),
-                Res::Overflow(_) => {
-                    Err(Error::overflow("BIGINT", &format!("({} / {})", lhs, rhs)).into())
-                }
-            },
-            Res::Truncated(_) => Err(Error::truncated().into()),
-            Res::Overflow(_) => {
-                Err(Error::overflow("DECIMAL", &format!("({} / {})", lhs, rhs)).into())
-            }
-        },
-        None => Ok(ctx.handle_division_by_zero().map(|()| None)?),
-    }
+    Ok(if result.is_truncated() {
```

Comment:
```go
	if err == types.ErrDivByZero {
		return 0, true, handleDivisionByZeroError(s.ctx)
	}
	if err == types.ErrTruncated {
		err = sc.HandleTruncate(errTruncatedWrongValue.GenWithStackByArgs("DECIMAL", c))
	}
	if err == types.ErrOverflow {
		newErr := errTruncatedWrongValue.GenWithStackByArgs("DECIMAL", c)
		err = sc.HandleOverflow(newErr, newErr)
	}
```
both truncated and overflow should be handle specificly

Reply:
Yes. We need to handle all the error cases while executing division of `Decimal`. However, no need to handle truncated case while casting `Decimal` to `i64` whose result is as same as `TiDB`.